exercises/1-1-getting-started/2-Using-venvs.md

```diff
@@ -0,0 +1,81 @@
+$ source myenv/bin/activate  # Activate the virtual environment
+(myenv) $ which python       # Get the current Python interpreter
+(myenv) $ pip list           # List the packages installed by default
+(myenv) $ deactivate         # Deactivate the virtual environment
+$ which python            # Get the current Python interpreter
+```
+
+## Using `virtualenv` (optional)
+
+[`virtualenv`](https://virtualenv.pypa.io/en/latest/) has many extra featuers on top of `venv`, but it is not part of the Python standard library. It is a standalone package that needs to be installed first (check [virtualenv installation](https://virtualenv.pypa.io/en/latest/installation.html)), or use `uv` like:
+```bash
+$ uv tool install virtualenv
+```
+
+Creating and activating virtual environments works very much like `venv`, but it features many other options:
+
+```bash
+$ virtualenv venv --help       # Get help and see the options
+$ which python                 # Get the current Python interpreter
+$ virtualenv myvirtualenv      # Create a new virtual environment
+$ source myvirtualenv/bin/activate  # Activate the virtual environment
+(myvirtualenv) $ which python       # Get the current Python interpreter
+(myvirtualenv) $ pip list           # List the packages installed by default
+(myvirtualenv) $ deactivate         # Deactivate the virtual environment
+$ which python                      # Get the current Python interpreter
+```
+
+## Using `uv venv`
+
+Using [`uv`](https://docs.astral.sh/uv/) to create virtual environments is as straightforward as expected:
+
+```bash
+uv venv --help       # Get help and see the options
+uv venv myuvenv      # Create a new virtual environment
+source myuvenv/bin/activate    # Activate the virtual environment
+(myuvenv) $ which python       # Get the current Python interpreter
+(myuvenv) $ uv pip list        # List the packages installed by default
+(myuvenv) $ deactivate         # Deactivate the virtual environment
+```
+
+------------------------------------------------------------
+**IMPORTANT:** Creating and activating Python virtual environments using `venv`, `virtualenv` or `uv` is quite similar and straightforward. However, note that the default packages installed in the newly created virtual environment may vary depending on the Python version, the creation options and the tool used to create the virtual environment. For example, `venv` and `virtualenv` install `pip` by default, while `uv` doesn't.
+
+In virtual environments using `pip` as package manager, it is recommended to always install the latest versions of `pip`, `setuptools` and `wheel` before any other package, to avoid installation problems.
+
+```bash
+$ python -m pip install --upgrade pip setuptools wheel
+```
+
+------------------------------------------------------------
+**EXERCISES:**
+
+1. Use `venv` to create a virtual environment called `.venv-1.26`, install NumPy version 1.26 on it with `pip`, and run a python command which.
```

Comment:
you mean `which python` command?

Reply:
The final part of the exercise was missing, thanks. Fixed.